components/director/pkg/cronjob/cronjob.go

```diff
@@ -47,22 +47,25 @@ func (r *cronJobRunner) Start(ctx context.Context) {
 
 	for {
 		start := time.Now()
+		log.C(ctx).Infof("Starting CronJob %s execution", r.CronJob.Name)
 		r.CronJob.Fn(newCtx)
-		select {
-		case <-newCtx.Done():
+		if newCtx.Err() != nil {
+			log.C(ctx).Infof("CronJob %s stopped due to context done", r.CronJob.Name)
```

Comment:
This seem to be erroneous "Done", not completion "Done".

Reply:
The function `context.Err` returns non-nil error if parent context is done or the cancel has been called. See [here](https://cs.opensource.google/go/go/+/refs/tags/go1.19.1:src/context/context.go;l=101-106)